src/hotspot/cpu/x86/nativeInst_x86.cpp

```diff
@@ -510,12 +510,28 @@ void NativeJump::check_verified_entry_alignment(address entry, address verified_
 //
 void NativeJump::patch_verified_entry(address entry, address verified_entry, address dest) {
```

Comment:
Should we assert that the appropriate lock is held here?

Reply:
This code can be called from different places and therefore can require different locks, so I wold prefer not to put any caller-specific logic to this function. Also, original 32bit code doesn't have such assert.